var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -0,0 +1,239 @@
+from spack import *
+import sys
+
+class Dealii(Package):
+    """C++ software library providing well-documented tools to build finite element codes for a broad variety of PDEs."""
+    homepage = "https://www.dealii.org"
+    url      = "https://github.com/dealii/dealii/releases/download/v8.4.0/dealii-8.4.0.tar.gz"
+
+    version('8.4.0', 'ac5dbf676096ff61e092ce98c80c2b00')
+    version('dev', git='https://github.com/dealii/dealii.git')
+
+    variant('mpi',      default=True,  description='Compile with MPI')
+    variant('arpack',   default=True,  description='Compile with Arpack and PArpack (only with MPI)')
+    variant('doxygen',  default=False, description='Compile with Doxygen documentation')
+    variant('hdf5',     default=True,  description='Compile with HDF5 (only with MPI)')
+    variant('metis',    default=True,  description='Compile with Metis')
+    variant('netcdf',   default=True,  description='Compile with Netcdf (only with MPI)')
+    variant('oce',      default=True,  description='Compile with OCE')
+    variant('p4est',    default=True,  description='Compile with P4est (only with MPI)')
+    variant('petsc',    default=True,  description='Compile with Petsc (only with MPI)')
+    variant('slepc',    default=True,  description='Compile with Slepc (only with Petsc and MPI)')
+    variant('trilinos', default=True,  description='Compile with Trilinos (only with MPI)')
+
+    # required dependencies, light version
+    depends_on ("blas")
+    depends_on ("boost",     when='~mpi')
+    depends_on ("boost+mpi", when='+mpi')
+    depends_on ("bzip2")
+    depends_on ("cmake")
+    depends_on ("lapack")
+    depends_on ("muparser")
+    depends_on ("suite-sparse")
+    depends_on ("tbb")
+    depends_on ("zlib")
+
+    # optional dependencies
+    depends_on ("mpi", when="+mpi")
+    depends_on ("arpack-ng+mpi", when='+arpack+mpi')
+    depends_on ("doxygen", when='+doxygen')
+    depends_on ("hdf5+mpi~cxx", when='+hdf5+mpi') #FIXME NetCDF declares dependency with ~cxx, why?
+    depends_on ("metis", when='+metis')
+    depends_on ("netcdf+mpi", when="+netcdf+mpi")
+    depends_on ("netcdf-cxx", when='+netcdf+mpi')
+    depends_on ("oce", when='+oce')
+    depends_on ("p4est", when='+p4est+mpi')
+    depends_on ("petsc+mpi", when='+petsc+mpi')
+    depends_on ("slepc", when='+slepc+petsc+mpi')
+    depends_on ("trilinos", when='+trilinos+mpi')
+
+    # developer dependnecies
+    #depends_on ("numdiff") #FIXME
+    #depends_on ("astyle") #FIXME
+
+    def install(self, spec, prefix):
+        options = []
+        options.extend(std_cmake_args)
+
+        # CMAKE_BUILD_TYPE should be DebugRelease | Debug | Release
+        for word in options[:]:
+            if word.startswith('-DCMAKE_BUILD_TYPE'):
+                options.remove(word)
+
+        dsuf = 'dylib' if sys.platform == 'darwin' else 'so'
+        options.extend([
+            '-DCMAKE_BUILD_TYPE=DebugRelease',
```

Comment:
DebugRelease means that it will build both versions, users will be able to do `make debug` or `make release` and different builds of the library will be used. `Debug` is not only about `-g`, but also contains a lot of Asserts. So in sense, yes, `DebugRelease` is its own build type.